Python/assemble.c

```diff
@@ -481,7 +481,11 @@ compute_localsplus_info(_PyCompile_CodeUnitMetadata *umd, int nlocalsplus,
         if (PyDict_Contains(umd->u_fasthidden, k)) {
             kind |= CO_FAST_HIDDEN;
         }
-        if (PyDict_GetItem(umd->u_cellvars, k) != NULL) {
+        int has_cell = PyDict_Contains(umd->u_cellvars, k);
+        if (has_cell < 0) {
+            return -1;
```

Comment:
Functions in this file return SUCCESS/ERROR.

While you are here, you can also add `if (PyErr_Occurred()) {return ERROR;}` after loops `while (PyDict_Next(...))`. The condition can be false in two cases: end of iteration and error.

And the existing `PyDict_Contains()` is not checked for error.

And `PyLong_AS_LONG()` can fail, unlike to `PyInt_AS_LONG()` in Python 2.

Reply:
I modified existing code to check for PyDict_Contains() and PyLong_AS_LONG() failures.

Reply:
With additions checks, I'm not sure that it's still worth it to check for PyErr_Occurred() after the loops.

Reply:
`PyDict_Next()` returns 0 on error. The only way to distinguish an error from the normal loop ending is to call `PyErr_Occurred()`.